Mathlib/Algebra/Order/Floor/Semiring.lean

```diff
@@ -0,0 +1,570 @@
+/-
+Copyright (c) 2018 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Kevin Kappelmann
+-/
+import Mathlib.Data.Nat.Cast.Order.Field
```

Comment:
Looking at an imports of this file, it seems the most important one is `Mathlib.Data.NNReal.Basic`, which does not require anything from `Mathlib.Data.Nat.Cast.Order.Field`. One alternative is to do a definitions + basics split, which I think might result in much nicer results downstream from `NNReal`.

Reply:
Let me see if I can get a prototype PR up.

Reply:
My alternative proposal is #23662. We currently get bigger reductions but only in a few files, mostly because `Algebra.Order.Archimedean.Basic` imports everything again.

We could also consider a hybrid approach where we have a `Defs` file like in #23662 and then lemmas files for `FloorSemiring` and `FloorRing`. What do you think?

Reply:
I would prefer the hybrid approach you put forth. I have had a preference towards even smaller files (at most 1000 lines) while I've been doing these splits, and I don't see it likely that some theorem will use both the `Nat` and `Int` versions of floor/ceiling. So the hierarchy is `Defs` being imported by `Semiring` and `Ring` files, and a separate file for the `FloorRingToSemiring` section.

If that is not fine with you I will be happy with #23662 as it is.

Reply:
@Vierkantor I would prefer the hybrid approach you put forth. I have had a preference towards even smaller files (at most 1000 lines) while I've been doing these splits, and I don't see it likely that some theorem will use both the `Nat` and `Int` versions of floor/ceiling. So the hierarchy is `Defs` being imported by `Semiring` and `Ring` files, and a separate file for the `FloorRingToSemiring` section.

If that is not fine with you I will be happy with #23662 as it is.

Reply:
@Vierkantor #23692